src/edu/csus/ecs/pc2/core/Utilities.java

```diff
@@ -957,61 +967,61 @@ public long stringToLongSecs(String s) {
 
-        if (serializedFile.isExternalFile()) {
-
-            String testFileName;
-
-            if (alternateCDPPath != null && alternateCDPPath.trim().length() > 0) {
-
-                testFileName = getSecretDataPath(alternateCDPPath, problem) + File.separator + serializedFile.getName();
-                if (fileExists(testFileName)) {
-                    return testFileName;
-                }
-            }
+        String testFileName = locateJudgesDataFile(problem, serializedFile.getName(), alternateCDPPath);
+        
+        if (testFileName != null && fileExists(testFileName)) {
+            return testFileName;
+        }
+        
+        testFileName = serializedFile.getAbsolutePath();
 
-            String secretPathPattern = File.separator + SECRET_DATA_DIR + File.separator;
-            String fullPathName = serializedFile.getAbsolutePath();
+        if (fileExists(testFileName)) {
+            return testFileName;
+        }
 
-            secretPathPattern = secretPathPattern.replace('\\', '.');
+        return null;
+    }
+    
+    public static String locateJudgesDataFile(Problem problem,  String baseFileName, String alternateCDPPath) {
 
-            if (fullPathName.matches(secretPathPattern)) {
+        String testFileName = null;
 
-                // return filename if source file under /data/secret/ somewhere
-                testFileName = getSecretDataPath(problem.getCCSfileDirectory(), problem) + File.separator + serializedFile.getName();
-                if (fileExists(testFileName)) {
-                    return testFileName;
-                }
-            }
+        if (alternateCDPPath != null && alternateCDPPath.trim().length() > 0) {
+            
+            // Try to find data file under secret directory
 
-            testFileName = alternateCDPPath + File.separator + problem.getShortName() + File.separator + serializedFile.getName();
+            testFileName = getSecretDataPath(alternateCDPPath, problem) + File.separator + baseFileName;
             if (fileExists(testFileName)) {
-                // return filename if under shortname/ path
                 return testFileName;
             }
+        }
+        
+        if (alternateCDPPath != null && alternateCDPPath.trim().length() > 0) {
+            
+            // Try to find data file under sample directory
 
-            testFileName = problem.getExternalDataFileLocation() + File.separator + serializedFile.getName();
+            testFileName = getSampleDataPath(alternateCDPPath, problem) + File.separator + baseFileName;
             if (fileExists(testFileName)) {
                 return testFileName;
             }
+        }
 
-            testFileName = serializedFile.getAbsolutePath();
+        testFileName = alternateCDPPath + File.separator + problem.getShortName() + File.separator + baseFileName;
+        if (fileExists(testFileName)) {
+            // try to find filename under shortname/ path
```

Comment:
This comment seems superfluous (left over from previous code versions?) It describes neither the operation associated with the (preceding) `if` statement, nor the (single) line of code which follows it.   Suggest either rewriting the comment to better explain whatever it is that it's trying to explain, or else deleting it.